### PR TITLE
Openings: remove reimport duplicates

### PR DESCRIPTION
**Pull request by sbusato**

### CODE REVIEW

File: ifrs17/Import/ImportStorage.ipynb [hunk 1]
```diff
@@ -224,7 +224,10 @@
                 "\n        var openingRawVariables = Enumerable.Empty<RawVariable>();",
                 "\n        var openingIfrsVariables = Enumerable.Empty<IfrsVariable>();",
                 "\n",
-                "\n        if(allImportScopes.Select(dn => DataNodeDataBySystemName[dn]).Any(dnd => dnd.Year < args.Year)) {",
+                "\n        var allImportScopesAtInceptionYear = allImportScopes.Select(dn => DataNodeDataBySystemName[dn]).Where(dnd => dnd.Year == args.Year).Select(x => x.DataNode).ToArray();",
```
Comment (dcolleoni) [inline ifrs17/Import/ImportStorage.ipynb:227]: make it an HashSet then Except is better performing

File: ifrs17/Import/ImportStorage.ipynb [hunk 2]
```diff
@@ -251,11 +254,15 @@
                 "\n        ",
                 "\n        //SetPartition to current Period",
                 "\n        await querySource.Partition.SetAsync<PartitionByReportingNodeAndPeriod>(TargetPartition);",
-                "\n        ",
-                "\n        if(primaryScope.Select(dn => DataNodeDataBySystemName[dn]).Any(dnd => dnd.Year == args.Year) && ImportFormat != ImportFormats.Opening)",
+                "\n",
+                "\n        var primaryScopeAtInceptionYear = allImportScopesAtInceptionYear.Where(dn => primaryScope.Contains(dn));",
```
Comment (dcolleoni) [inline ifrs17/Import/ImportStorage.ipynb:258]: These are two hashset. Would an Intersect perform better? @dtrzesniak

**Review by dcolleoni — APPROVED** [review]

File: ifrs17/Import/ImportStorage.ipynb (hunk 1 above)
Comment (sbusato) [reply ifrs17/Import/ImportStorage.ipynb]: Done.

File: ifrs17/Import/ImportStorage.ipynb (hunk 2 above)
Comment (sbusato) [reply ifrs17/Import/ImportStorage.ipynb:258]: After a brief research, it seems that Contains performs better than Intersect when working with Hashsets. @dtrzesniak @dcolleoni

File: ifrs17/Import/ImportStorage.ipynb [hunk 3]
```diff
@@ -251,11 +254,15 @@
...
+                "\n        var secondaryScopeAtInceptionYear = allImportScopesAtInceptionYear.Where(dn => secondaryScope.Contains(dn));        ",
```
[the remark this answers is not in the document]
Comment (sbusato) [reply ifrs17/Import/ImportStorage.ipynb:259]: Same as above.

**Pull request merged**